bstats-bukkit-lite/src/main/java/org/bstats/bukkit/MetricsLite.java

```diff
@@ -233,24 +235,45 @@ private JSONObject getServerData() {
      * Collects the data and sends it afterwards.
      */
     private void submitData() {
-        final JSONObject data = getServerData();
+        final JsonObject data = getServerData();
 
-        JSONArray pluginData = new JSONArray();
+        JsonArray pluginData = new JsonArray();
         // Search for all other bStats Metrics classes to get their plugin data
         for (Class<?> service : Bukkit.getServicesManager().getKnownServices()) {
             try {
                 service.getField("B_STATS_VERSION"); // Our identifier :)
 
                 for (RegisteredServiceProvider<?> provider : Bukkit.getServicesManager().getRegistrations(service)) {
                     try {
-                        pluginData.add(provider.getService().getMethod("getPluginData").invoke(provider.getProvider()));
+                        Object plugin = provider.getService().getMethod("getPluginData").invoke(provider.getProvider());
+                        if (plugin instanceof JsonObject) {
+                            pluginData.add((JsonObject) plugin);
+                        } else { // old bstats version compatibility [UNTESTED]
+                            try {
+                                Class<?> jsonObjectJsonSimple = Class.forName( "org.json.simple.JSONObject" );
+                                if ( plugin.getClass().isAssignableFrom( jsonObjectJsonSimple ) ) {
+                                    Method jsonStringGetter = jsonObjectJsonSimple.getDeclaredMethod( "toJSONString" );
+                                    jsonStringGetter.setAccessible( true );
+                                    String jsonString = (String) jsonStringGetter.invoke( plugin );
+                                    JsonObject object = new JsonParser().parse( jsonString ).getAsJsonObject();
+                                    pluginData.add( object );
+                                }
+                            } catch ( ClassNotFoundException exc ) {
+                                // minecraft version 1.14+
+                                String minecraftVersion = Bukkit.getBukkitVersion().split( "-" )[0];
+                                this.plugin.getLogger().severe( "[bStats Metrics] Usage of old bStats version found on minecraft " + minecraftVersion +
+                                        ". Update bStats to version 1.5 . Older minecraft version compatibility won't be lost! " +
+                                        "Server owners, please send this to the developer of the plugin." );
```

Comment:
I don't think this message in necessary. If there is a plugin that uses `JSONObject`, it will throw an exception immediately on plugin start anyway. Maybe we can just add a small debug message when `logFailedRequests` is enabled, just to be on the safe side:
```java
this.plugin.getLogger().severe("Encountered unexpected exception", e);
```